src/fragments/lib/graphqlapi/js/working-with-files.mdx

```diff
@@ -0,0 +1,984 @@
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+async function deleteImageForCurrentSong() {
+  if (!currentSong) return;
+
+  const _song = await getSong();
+
+  if (!_song?.coverArtKey) return;
+
+  await removeImageFromSong();
+
+  try {
+    // Delete the file from S3:
+    const deletedImage = await Storage.remove(_song?.coverArtKey);
+  } catch (error) {
+    console.error("Error deleting image: ", error);
+  }
+}
+``` 
+
+</Block>
+<Block name="JavaScript">
+
+```js
+async function deleteImageForCurrentSong() {
+  if (!currentSong) return;
+
+  const _song = await getSong();
+
+  if (!_song?.coverArtKey) return;
+
+  await removeImageFromSong();
+
+  try {
+    // Delete the file from S3:
+    const deletedImage = await Storage.remove(_song?.coverArtKey);
+  } catch (error) {
+    console.error("Error deleting image: ", error);
+  }
+}
+```
+
+</Block>
+</BlockSwitcher>
+
+### Deleting the record and associated file
+
+When deleting a record, we may also want to delete the associated file:
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+// Deletes current song. If song has an image, deletes image from Storage:
+async function deleteCurrentSong() {
+  if (!currentSong) return;
+
+  const _song = await getSong();
+
+  // Save the file key for deleting the file after the record is deleted:
+  const currentSongImageKey = _song?.coverArtKey;
+
+  const songDetails: DeleteSongInput = {
+    id: currentSong.id,
+  };
+
+  try {
+    const deletedSong = await API.graphql<GraphQLQuery<DeleteSongMutation>>({
+      query: mutations.deleteSong,
+      variables: { input: songDetails },
+    });
+
+    setCurrentSong(null);
+
+    if (!currentSongImageKey) return;
+
+    try {
+      const deletedImage = await Storage.remove(currentSongImageKey);
+      clearLocalState();
+    } catch (error) {
+      console.error("Error deleting image: ", error);
+    }
+  } catch (error) {
+    console.error("Error deleting song: ", error);
+  }
+}
+``` 
+
+</Block>
+<Block name="JavaScript">
+
+```js
+// Deletes current song. If song has an image, deletes image from Storage:
+async function deleteCurrentSong() {
+  if (!currentSong) return;
+
+  const _song = await getSong();
+
+  // Save the file key for deleting the file after the record is deleted:
+  const currentSongImageKey = _song?.coverArtKey;
+
+  const songDetails = {
+    id: currentSong.id,
+  };
+
+  try {
+    const deletedSong = await API.graphql({
+      query: mutations.deleteSong,
+      variables: { input: songDetails },
+    });
+
+    setCurrentSong(null);
+
+    if (!currentSongImageKey) return;
+
+    try {
+      const deletedImage = await Storage.remove(currentSongImageKey);
+      clearLocalState();
+    } catch (error) {
+      console.error("Error deleting image: ", error);
+    }
+  } catch (error) {
+    console.error("Error deleting song: ", error);
+  }
+}
+```
+
+</Block>
+</BlockSwitcher>
+
+## Configuring authorization
```

Comment:
I think re: my comment on the very top. This section can probably be collapsed into a warning that's at the very top of this page.